### ISSUE TITLE
Input Redirection stops responding after variable soft resets

### ISSUE THREAD

**Issue by cursedtoast** [LumaTeam/Luma3DS#1195]
**System model:**

New 2DS

**SysNAND version (+emu/redNAND version if applicable):**

SysNAND 11.8.0

**Entrypoint (How/what you're using to boot Luma3DS):**

Boot9Strap

**Luma3DS version:**

9.1

--

Autoboot EmuNAND: ( )

Use EmuNAND FIRM if booting with R: ( )

Enable loading external FIRMs and modules: ( )

Enable game patching: ( )

Show NAND or user string in System Settings: (x)

Show GBA boot screen in patched AGB_FIRM: ( )

Patch ARM9 access: ( )

Set developer UNITINFO: ( )

Disable ARM11 exception handlers: ( )

--

**Explanation of the issue:**

I've written a shiny hunting bot for Gen 7 Pokemon games that takes advantage of the Rosalina Input Redirection.

Step forward -> If not shiny -> soft reset.

The problem is, randomly, Input Redirection stops working. And interestingly, it's always right after loading a game in Pokemon Ultra Sun/Moon. The 3ds isn't locked up itself. I can try stopping the client on the PC (currently using Release 3.0), and restarting it, with no go. Even starting a new client on a different PC I can't get it to work.

I've also tried disabling Input Redirection on the 3ds system, then re-enabling it.

The only way I can get Input Redirection to work again, is to turn off the 3ds and turn it back on and start all over again.

There isn't a specific time this happens. It can happen 900 soft resets in, it can happen 40 soft resets in. It might not ever happen at all during a hunt. Though usually, it happens around 600 soft resets in. The program I wrote literally just emulates an Xbox 360 controller. Which is still functioning fully when this issue happens.

Something is happening where Input Redirection stops responding and connections to it are no longer possible until after system restart. Unfortunately it's not consistent, but it does happen several times a day and is annoying.

I've noticed when this happens, if I press the home menu, the 3ds internet icon is set to no internet connection and takes a second to connect again. When the issue isn't happening, pressing the home menu the internet icon stays solid. 

So I think something is causing the wireless to drop on game load at some point? Tried several different routers and I have a stable connection, so I know it's not just losing internet connection due to poor connection quality.

**Steps to reproduce:**

1. Using Input Redirection, load your save file in Pokemon Ultra Sun/Moon.

2. Soft reset

3. Repeat steps 1-2 a variable amount of times. Eventually, upon game load, Input Redirection will stop communicating with any clients and the DS system will need to be rebooted and Input Redirection restarted in order to receive communication from any client again.

**Comment by Bappsack** [comment]
its a common Issue with the Luma Debugger, once Wifi Dies the Game freeze, only way to recover is disable the Luma Debugger in Rosalina Menu

**Comment by cursedtoast** [comment]
I don't think I ever have the Rosalina debugger enabled in the first place - does it enable when you enable Input Redirection?

**Comment by Bappsack** [comment]
Yes

cursedtoast <notifications@github.com> schrieb am Mi., 17. Okt. 2018, 02:30:

**Comment by cursedtoast** [comment]
Okay, I'll take a look the next time my bot locks up. Should I just disable the debugger in the future to prevent this from happening, then? Thank you.

**Comment by Bappsack** [comment]
If u disable it, IR won't work anymore. I would say use NTR instead

cursedtoast <notifications@github.com> schrieb am Mi., 17. Okt. 2018, 03:01:

**Comment by cursedtoast** [comment]
The Luma Debugger showed as already disabled, after I restarted my 3ds,  and enabled Input Redirection.

If you mean the NTR Input Redirection, I looked into that as well, but apparently NTR has an issue where on soft reset, the system will randomly freeze. I can get up to about 600-1000 SR's usually before issue, these days, where I guess for NTR it freezes much quicker. Unless the issue with NTR I was reading about has changed.
